client/client.go

```diff
@@ -322,6 +322,9 @@ func (c *Client) LogsWebSocket(container string, entries int) (SocketReader, err
 	if err == websocket.ErrBadHandshake {
 		return nil, fmt.Errorf("websocket handshake failed with status %d", resp.StatusCode)
 	}
+	if err != nil {
+		return nil, err
```

Comment:
we could include more context here (although I know most of the codebase is a little older and doesn't do this, it's still generally good practice), ie:

```suggestion
		return nil, fmt.Errorf("failed to connect to daemon at %s: %s", url.Host, err.Error())
```